Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Composable/NolPay/NolPayLinkCardComponent.swift

```diff
@@ -317,16 +317,20 @@ public class NolPayLinkCardComponent: PrimerHeadlessCollectDataComponent {
                                                                                 phoneVendor: "Apple",
                                                                                 phoneModel: UIDevice.modelIdentifier!)
             let client = PrimerAPIClient()
-            
-            return try await withCheckedThrowingContinuation { continuation in
-                client.fetchNolSdkSecret(clientToken: clientToken, paymentRequestBody: requestBody) { result in
-                    switch result {
-                    case .success(let appSecret):
-                        continuation.resume(returning: appSecret.sdkSecret)
-                    case .failure(let error):
-                        continuation.resume(throwing: error)
+            if #available(iOS 13, *) {
```

Comment:
This hurts my eyes, and it feels like it should already be guaranteed by `#if canImport(PrimerNolPaySDK)`, but preprocessor and compiler are not communicating here

Reply:
This can all be replaced with PromiseKit promises, but best left for a different PR